### PR TITLE
Add literature to step-35

### PR DESCRIPTION
**Pull request by masterleinad**
In response to #2897 add the relevant references.

### CODE REVIEW

File: examples/step-35/doc/intro.dox [hunk 1]
```diff
@@ -143,6 +142,33 @@ projection methods that we have just described:
...
+<ul>
+  <li> J. van Kan, A second-order accurate pressure-correction scheme for viscous incompressible flow, SIAM J. Sci. Stat. Comput. 7 (3) (1986) 870–891
+</ul>
+and is analyzed by Guermond in
+<ul>
+  <li> J.-L. Guermond, Un résultat de convergence d’ordre deux en temps pour
+                       l’approximation des équations de Navier–Stokes par une technique de projection incrémentale,
+       ESAIM: Mathematical Modelling and Numerical Analysis, vol. 33, no. 01, pp. 169–189, 1999
+</ul>
+for the case \f$\nu = 1\f$.
```
Comment (bangerth) [inline examples/step-35/doc/intro.dox]: No need for `\f` -- we automatically expand the dollar sign to the form you use.

File: examples/step-35/doc/intro.dox [hunk 2]
```diff
@@ -143,6 +142,33 @@ projection methods that we have just described:
...
+<p>
+The standard incremental scheme and the rotational incremental scheme were first considered by van Kan in
+<ul>
+  <li> J. van Kan, A second-order accurate pressure-correction scheme for viscous incompressible flow, SIAM J. Sci. Stat. Comput. 7 (3) (1986) 870–891
+</ul>
+and is analyzed by Guermond in
+<ul>
+  <li> J.-L. Guermond, Un résultat de convergence d’ordre deux en temps pour
+                       l’approximation des équations de Navier–Stokes par une technique de projection incrémentale,
+       ESAIM: Mathematical Modelling and Numerical Analysis, vol. 33, no. 01, pp. 169–189, 1999
```
Comment (bangerth) [inline examples/step-35/doc/intro.dox]: This will be printed without endlines. Want to put the title of the paper in quotes? Same above and below.

Comment (masterleinad) [inline examples/step-35/doc/intro.dox]: Thanks!

**Comment by bangerth** [comment]
Thanks, @masterleinad !

**Pull request merged**